db/versions/0017.yml

```diff
@@ -0,0 +1,242 @@
+          (properties ->> 'workerPoolId')::text as worker_pool_id,
+          (properties ->> 'reported')::timestamptz as reported,
+          (properties ->> 'kind')::text as kind,
+          (properties ->> 'title')::text as title,
+          (properties ->> 'description')::text as description,
+          entity_buf_decode(properties, 'extra')::jsonb as extra,
+          public.gen_random_uuid() as etag
+        into new_row;
+        update worker_pool_errors
+        set (
+          error_id,
+          worker_pool_id,
+          reported,
+          kind,
+          title,
+          description,
+          extra,
+          etag
+        ) = (
+          new_row.error_id,
+          new_row.worker_pool_id,
+          new_row.reported,
+          new_row.kind,
+          new_row.title,
+          new_row.description,
+          new_row.extra,
+          new_row.etag
+        )
+        where
+          worker_pool_errors.error_id = decode_string_key(wmworker_pool_errors_entities_modify.partition_key) and
+          worker_pool_errors.etag = wmworker_pool_errors_entities_modify.old_etag;
+
+        if found then
+          return query select new_row.etag;
+          return;
+        end if;
+
+        perform worker_pool_errors.etag from worker_pool_errors
+        where worker_pool_errors.error_id = decode_string_key(wmworker_pool_errors_entities_modify.partition_key);
+
+        if found then
+          raise exception 'unsuccessful update' using errcode = 'P0004';
+        else
+          raise exception 'no such row' using errcode = 'P0002';
+        end if;
+      end
+  wmworker_pool_errors_entities_scan:
+    description: See taskcluster-lib-entities
+    mode: read
+    serviceName: worker_manager
+    args: pk text, rk text, condition text, size integer, page integer
+    returns: table (partition_key text, row_key text, value jsonb, version integer, etag uuid)
+    body: |-
+      declare
+      begin
+        if not condition is null then
+          raise exception 'condition not supported';
+        end if;
+        return query
+          select
+            wmworker_pool_errors_entities_scan.partition_key,
+            wmworker_pool_errors_entities_scan.row_key,
+            entity_buf_encode(
+              jsonb_build_object(
+                'PartitionKey', encode_string_key(error_id),
+                'RowKey', encode_string_key(error_id),
+                'errorId', error_id,
+                'workerPoolId', worker_pool_id,
+                'reported', reported,
+                'kind', kind,
+                'title', title,
+                'description', description),
+              'extra', extra::text) as value,
+            1 as version,
+          worker_pool_errors.etag as etag
+          from worker_pool_errors
+          where
+            partition_key is null or
+            error_id = decode_string_key(partition_key)
+          order by error_id
+          limit size + 1
+          offset page;
```

Comment:
If size or offset are not defined then something may go wrong here. One way to fix that is to do:

```
        limit case
          when (size is not null and size > 0) then size + 1
          else null
        end
        offset case
          when (page is not null and page > 0) then page
          else 0
        end;
```